scipy/signal/tests/test_signaltools.py

```diff
@@ -2800,6 +2800,188 @@ def test_hilbert2_types(self, dtype):
         assert_equal(np.real(signal.hilbert2(in_typed)).dtype, dtype)
 
 
+class TestEnvelope:
+    """Unit tests for function ._signaltools.envelope()`. """
+
+    @classmethod
+    def assert_close(cls, actual, desired, msg):
+        """Little helper to compare to arrays with proper tolerances"""
+        assert_allclose(actual, desired, atol=1e-12, rtol=1e-12, err_msg=msg)
```

Comment:
we should probably use `xp_assert_close` to make life easier in the future

Reply:
Fixed.